File: Nogs IDE.cpp
/*
 *
 *
 *  Copyright (C) Nogs GmbH, Andre Riesberg
 *  
 *  This program is free software; you can redistribute it and/or
 *  modify it under the terms of the GNU General Public License
 *  as published by the Free Software Foundation; either version 2
 *  of the License, or (at your option) any later version.
 *  
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *  
 *  You should have received a copy of the GNU General Public License
 *  along with this program; if not, 
 *  write to the Free Software Foundation, Inc.,
 *  51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
//---------------------------------------------------------------------------
#include <vcl.h>
#pragma hdrstop
#include <tchar.h>
//---------------------------------------------------------------------------
#include <Vcl.Styles.hpp>
#include <Vcl.Themes.hpp>














































USEFORM("SetupAllDlg.cpp", DialogSetupAll);
USEFORM("Wizards\ZeroConf\ZeroConfWizardDlg.cpp", DialogZeroConfWizard);
USEFORM("Wizards\ZeroConf\ZeroConfWizardAdaperFrm.cpp", FrameZeroConfWizardAdapter); /* TFrame: File Type */
USEFORM("SetupOneDlg.cpp", DialogSetupOne);
USEFORM("MainFrm.cpp", FormMain);
USEFORM("Dialogs\LiveCodingTargetsDlg.cpp", DialogLiveCodingTargets);
USEFORM("Dialogs\MemoryTrackerDlg.cpp", DialogMemoryTracker);
USEFORM("Dialogs\EtherBootProgrammerDlg.cpp", DialogEtherBootProgrammer);
USEFORM("Dialogs\CommunicationTestDlg.cpp", DialogCommunicationTest);
USEFORM("Dialogs\ModelessBaseDlg.cpp", DialogModeless);
USEFORM("Dialogs\ZeroConfDlg.cpp", DialogZeroConf);
USEFORM("Dialogs\TargetSettingsDlg.cpp", DialogTargetSettings);
USEFORM("Dialogs\ProjectDlg.cpp", DialogProject);
USEFORM("Dialogs\ProjectNewDlg.cpp", DialogProjectNew);
USEFORM("..\..\Librarys\CBuilderUC\MessageBoxDlg\MessageBoxDlg.cpp", DialogMessageBox);
USEFORM("Frames\TargetsFrm.cpp", FrameTargets); /* TFrame: File Type */
USEFORM("Frames\TopologieEditorFrm.cpp", FrameTopolgieEditor); /* TFrame: File Type */
USEFORM("Frames\SysLog.cpp", FrameSysLog); /* TFrame: File Type */
USEFORM("Frames\Setup\SetupBaseFrm.cpp", FrameSetupBase); /* TFrame: File Type */
USEFORM("Frames\Setup\SetupSysLogFrm.cpp", FrameSetupSysLog); /* TFrame: File Type */
USEFORM("Frames\ProjectFrm.cpp", FrameProject); /* TFrame: File Type */
USEFORM("EtherBootFrm.cpp", FrameEtherBoot); /* TFrame: File Type */
USEFORM("Frames\LuaEditorFrm.cpp", FrameLuaEditor); /* TFrame: File Type */
USEFORM("Frames\NogsDesigner.cpp", FrameDesigner); /* TFrame: File Type */
USEFORM("Frames\LogFrm.cpp", FrameLog); /* TFrame: File Type */
USEFORM("Frames\BrowserFrm.cpp", FrameBrowser); /* TFrame: File Type */
USEFORM("Frames\CommunicationHopFrm.cpp", FrameCommunicationHop); /* TFrame: File Type */
//---------------------------------------------------------------------------
WINAPI _tWinMain(HINSTANCE, HINSTANCE, LPTSTR, int)
{
  try
  {
     Application->Initialize();
     Application->MainFormOnTaskBar = true;
     TStyleManager::TrySetStyle("Carbon");
     Application->CreateForm(__classid(TFormMain), &FormMain);
     Application->Run();
  }
  catch (Exception &exception)
  {
     Application->ShowException(&exception);
  }
  catch (...)
  {
     try
     {
       throw Exception("");
     }
     catch (Exception &exception)
     {
       Application->ShowException(&exception);
     }
  }
  return 0;
}
//---------------------------------------------------------------------------

File: Messe_EVE/NOGS-IDE.lua
--[[
 *
 *
 *  Copyright (C) Nogs GmbH, Andre Riesberg
 *  
 *  This program is free software; you can redistribute it and/or
 *  modify it under the terms of the GNU General Public License
 *  as published by the Free Software Foundation; either version 2
 *  of the License, or (at your option) any later version.
 *  
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *  
 *  You should have received a copy of the GNU General Public License
 *  along with this program; if not, 
 *  write to the Free Software Foundation, Inc.,
 *  51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
--]]


local function execute(source)
  print("NOGS_IDE: Execute("..source..")")
  local app, error = load(source)
  source = nil
  if app then
    local success, result = pcall(app)
    if success then
      print('NOGS_IDE: Success', result)
      if result == nil then
        return '["Success"]'
      else
        return '[{"Success":"'..string.gsub(result, '"', '\\"')..'"}]'
      end
    else
      print('NOGS_IDE: Error during pcall', result)
      return '[{"error":"'..string.gsub(result,'%"',' ')..'"}]'
    end
  else
    print('NOGS_IDE: Error during load:',error)
    return '[{"error":"'..string.gsub(error, '%"',' ')..'"}]'
  end
end


function Port11111Handler(socket, ip, port, events, status)
  --print("Port11111Recv", socket, ip, port, events, status)
  sys.w5500.open(socket)
  while true do
    coroutine.yield()
    local recv, peerIP = sys.w5500.recv(socket);
    if recv then
      sys.w5500.send(socket, execute(recv), peerIP, 11110)
    end
  end
end


local socket = sys.w5500.create("UDP", 11111, coroutine.create(Port11111Handler))

